alacritty_terminal/src/term/mod.rs

```diff
@@ -1930,6 +1930,83 @@ mod tests {
     use crate::selection::{Selection, SelectionType};
     use crate::term::cell::{Cell, Flags};
 
+    #[test]
+    fn scroll_display_page_up() {
+        let size = SizeInfo::new(5., 10., 1.0, 1.0, 0.0, 0.0, false);
+        let mut term = Term::new(&MockConfig::default(), size, ());
+
+        // Create 20 lines of scrollback.
+        for _ in 0..29 {
+            term.newline();
+        }
+
+        term.scroll_display(Scroll::PageUp);
+        assert_eq!(term.vi_mode_cursor.point, Point::new(Line(-1), Column(0)));
+        assert_eq!(term.grid.display_offset(), 10);
+
+        term.scroll_display(Scroll::PageUp);
+        assert_eq!(term.vi_mode_cursor.point, Point::new(Line(-11), Column(0)));
+        assert_eq!(term.grid.display_offset(), 20);
+
+        term.scroll_display(Scroll::PageUp);
+        assert_eq!(term.vi_mode_cursor.point, Point::new(Line(-11), Column(0)));
+        assert_eq!(term.grid.display_offset(), 20);
```

Comment:
This tests for a perfect match, where a jump ends up in exactly the right spot with a full page.

I feel like it might be useful to have the scrollback buffer uneven, just to test the case where it can only jump *less* than a full page until it lands at the top?

Same applies to `PageDown` this should be trivial to add.